photutils/psf/photometry.py

```diff
@@ -275,10 +280,21 @@ def do_photometry(self, image, init_guesses=None):
                 sources['aperture_flux'] = aperture_photometry(
                     image, apertures)['aperture_sum']
 
-                init_guesses = Table(names=['x_0', 'y_0', 'flux_0'],
-                                     data=[sources['xcentroid'],
-                                           sources['ycentroid'],
-                                           sources['aperture_flux']])
+                # init_guesses should be the initial 3 required parameters --
+                # x, y, flux -- and then concatentated with any additional
+                # sources, if there are any
+                if self._output_cols is None:
+                    init_guesses = Table(names=['x_0', 'y_0', 'flux_0'],
+                                         data=[sources['xcentroid'],
+                                               sources['ycentroid'],
+                                               sources['aperture_flux']])
+                else:
+                    init_guesses = Table(names=['x_0', 'y_0', 'flux_0'] +
+                                         list(self._output_cols),
+                                         data=[sources['xcentroid'],
+                                               sources['ycentroid'],
+                                               sources['aperture_flux']] +
+                                         [sources[o] for o in self._output_cols])
```

Comment:
instead of the `if`/`else` I wonder if this is better done as leaving the original implementation and follow it with something like:
```
if self._output_cols is not None:
    for col_nm in self._output_cols:
        init_guesses[col_nm] = sources[col_nm]
    init_guesses[o] for o 
```